x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -406,6 +423,25 @@ protected LogicalPlan rule(Limit limit) {
          */
         private static Limit descendantLimit(UnaryPlan unary) {
             UnaryPlan plan = unary;
+            while (plan instanceof Aggregate == false) {
+                if (plan instanceof Limit limit) {
+                    return limit;
+                } else if (plan instanceof MvExpand) {
```

Comment:
Why not apply the same behavior for Aggregate - stop searching for a limit once an MvExpand is found?